framework/src/geomsearch/PenetrationLocator.C

```diff
@@ -127,7 +127,13 @@ PenetrationLocator::detectPenetration()
 void
 PenetrationLocator::reinit()
 {
+  // Delete the PenetrationInfo objects we own before clearing the
+  // map, or we have a memory leak.
+  for (auto & it : _penetration_info)
+    delete it.second;
```

Comment:
I have no problem changing pointers with limited scope immediately. The only way we'll ever get all these raw pointers out of here is by doing them one at a time. I have several PRs that do nothing but change pointers. I like using `unique_ptr` whenever we can (which is more often than people recognize) because generally the lifetime of the object is longer than the method that uses it so when promoting the pointer, you can update the interface to take a reference to the object instead. Anyway, feel free to go nuts.